packages/core/src/libraries/verification-status.ts

```diff
@@ -0,0 +1,59 @@
+import { nanoid } from 'nanoid';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import type Queries from '#src/tenants/Queries.js';
+import assertThat from '#src/utils/assert-that.js';
+
+export const expirationTimeout = 10 * 60 * 1000; // 10 minutes.
+
+export type VerificationStatusLibrary = ReturnType<typeof createVerificationStatusLibrary>;
+
+export const createVerificationStatusLibrary = (queries: Queries) => {
+  const {
+    findVerificationStatusByUserIdAndSessionId,
+    findVerificationStatusesByUserIdAndSessionId,
+    insertVerificationStatus,
+    deleteVerificationStatusesByIds,
+  } = queries.verificationStatuses;
+
+  const createVerificationStatus = async (userId: string, sessionId: string) => {
+    // Remove existing verification statuses for current user in current session.
+    const verificationStatuses = await findVerificationStatusesByUserIdAndSessionId(
+      userId,
+      sessionId
+    );
+
+    if (verificationStatuses.length > 0) {
+      await deleteVerificationStatusesByIds(verificationStatuses.map(({ id }) => id));
+    }
+
+    // When creating new verification record, we use session ID to identify the client device.
+    // The session ID is a cookie value, which is unique for each client.
+    // This prevents the user from proceeding after being verified on another device.
+    return insertVerificationStatus({
+      id: nanoid(),
```

Comment:
nit: we are now using `generateStandardId()` instead of `nanoid()`?

Reply:
OK, I wasn't aware of this. Thanks for the info